### ISSUE TITLE
CMake should not look for dependencies when all tests/examples/etc are disabled

### ISSUE THREAD

**Issue by yurivict** [thrust/thrust#1211]
```-DTHRUST_DEVICE_SYSTEM=CPP``` works, but ``` THRUST_MULTICONFIG_ENABLE_SYSTEM_CUDA=OFF``` doesn't.

**Comment by allisonvacanti** [comment]
The `MULTICONFIG` options should only matter when `THRUST_ENABLE_MULTICONFIG` is enabled. If `THRUST_ENABLE_MULTICONFIG` is disabled, it shouldn't matter what multiconfig systems are enabled.

What are the full set of CMake options used here?

**Comment by yurivict** [comment]
Indeed, with ```THRUST_ENABLE_MULTICONFIG=ON``` it works.
```
CMAKE_OFF=      THRUST_ENABLE_HEADER_TESTING THRUST_ENABLE_TESTING THRUST_ENABLE_EXAMPLES THRUST_MULTICONFIG_ENABLE_SYSTEM_CUDA
CMAKE_ON=       THRUST_MULTICONFIG_ENABLE_SYSTEM_CPP THRUST_ENABLE_MULTICONFIG
```

But why check this at all when ```thrust``` is a headers-only library and tests and examples are disabled? Or does it hard-code these choices in installed headers?

**Comment by allisonvacanti** [comment]
The short answer is that Thrust's CMake build system is a work in progress. Right now it's mainly being used by developers, so it lacks many things that packagers would want (as you've noticed!).

These options are not hardcoded into the headers and could be disabled when no build targets are generated. I'll look into this at some point, it should be an easy fix.

**Comment by allisonvacanti** [comment]
Fixed in #1247.
